book/src/developer-guides/frame-nft.md

```diff
@@ -0,0 +1,52 @@
+# Overview
+Frame-NFT is aimed at developers to provide them with a reliable and adjustable codebase for NFT-based projects. 
+This is especially necessary due to the various forms that NFTs can present as, each with their own unique traits.
+In the following we will do a quick recap of what makes NFTs special followed by how that is reflected in Frame-NFT's code.
+And finally, we will discuss two use cases.
+
+---
+
+# Recap
+NFTs (Non-Fungible-Tokens) have a few characteristics setting them apart from fungible ones like currency, precious metals, or carbon credits:
+* Non - Interchangeable 
+* Specific creator and owner
+* Unique digital signature
+* Unique asset like Art, Music or Property
+* Provide complete history of their creation and trades
+
+They represent ownership of a unique tangible or intangible asset, and record its history on the blockchain.
+
+---
+
+# Workflow
+With the nature of NFTs in mind, it becomes imperative to have a straightforward implementation aiming to accommodate their traits.
+We achieve this by having multiple methods to define distinct types and characteristics in an asset.
+Ownership being a core concept in NFTs makes the presentation and trading of ownership a vital part of how we interact with them.
+This makes storing and searching for specific information an essential part of this framework.
+To that end, we leverage BTreeMaps as they reference data stored in them using BTreeSets consisting of two vectors.
+Safety is key when it comes to processing of data, which is why we use Blake2-128Concat to hash all inputs and storage.
```

Comment:
This is a bit too much implementation detail for a README, as well as that it still doesn't really reflect how a developer uses the pallet